at_parse.c

```diff
@@ -403,7 +403,7 @@ EXPORT_DEF int at_parse_cmgr(char *str, size_t len, int *tpdu_type, char *sca, s
 			/* tpdu_parse_deliver sets chan_dongle_err */
 			return -1;
 		}
-		res = ucs2_to_utf8(msg16_tmp, res, msg, res * 2 + 2);
+		res = ucs2_to_utf8(msg16_tmp, res, msg, res * 4 + 2);
```

Comment:
`+ 1` then?

Reply:
I don't know why there was "+ 2". The "res * 4" already makes some assumptions about the buffer being passed is large enough. It doesn't hurt to have some extra space though ...

Reply:
> It doesn't hurt to have some extra space though ...

Extra space for a single NUL, if you're unsure: sure. But two extra spaces? That hurts, because it implies that that space is used for something.

The `X*2+2` calculation looks like you're converting from 1-octet to 2-octet chars keeping room for a 2-octet NUL, so that made sense, even though it was wrong here. But I cannot think of a logical reason to do `X*4+2`.